hooks/schema-generation.js

```diff
@@ -0,0 +1,41 @@
+const {TypeScriptGenerator, FormatHelpers} = require('@asyncapi/generator-model-sdk');
+const fs = require('fs');
+const Path = require('path');
+
+const preset = {
+  interface: {
+    async property({ renderer, propertyName, property }) {
+      const name = propertyName;
+      const signature = renderer.renderTypeSignature(property, { isRequired: renderer.model.isRequired(propertyName) });
+      return `${name}${signature};`;
+
+    }
+  }
+}
+/**
+ * Use AsyncAPI generator model gen library to generate all schemas.
+ */
+module.exports = {
+  'generate:after': async (generator) => {
+    const typescriptGenerator = new TypeScriptGenerator({modelType: "interface", presets: [preset]});
+    const generatedModels = await typescriptGenerator.generate(generator.asyncapi);
+    const targetDir = Path.join(generator.targetDir, 'src/schemas/');
+    for (const generatedModel of generatedModels) {
+      const fileContent = `
+${generatedModel.model.getImmediateDependencies().map((value) => {return FormatHelpers.toPascalCase(value)}).join('\n')}
```

Comment:
What 😄 As I remember this function was called `getDependsOn` wtf :) Mangela's effect?

Reply:
😄 I changed the name in the PR because this explains exactly what it is `getting the immediate dependencies from the model` which means you dont get all the dependencies recursively for nested types 😄 At least that was the thought

Reply:
For me `getDedendsOn` also means this same. I'm afraid that the name may be incomprehensible to the community.

Reply:
Feel free to create a PR for the change, this was just my thoughts, we can get a third part to review it 😄

Reply:
To me, `getDependsOn` doesn't even make sense in English. You're chaining two verbs: "get" and "depends". You don't "get" a verb, you get "something". I think the "getImmediateDependencies" name is fine. It makes it explicit it's only "immediate", whatever that means.

Reply:
@fmvilas Yes, exactly, only `immediate` may cause problems with understanding. `getDependencies` cannot be used, because there we will have also problems - JSON Schema has `dependencies` property - maybe `dependsOn`? :)

Reply:
I don't have enough context on the real meaning of this, that's why I said: "whatever that means". Seeing the word "immediate" will make me suspect there's something special I'll have to check. If I see `dependsOn`, I still can think it will get all the deps, including the nested ones.

Reply:
`dependsOn` sounds more like you are checking `does this model depend on this other model` 😄 

By `Immediate` I mean the closest/nearest dependencies for that model.

Reply:
So maybe `getNearestDependencies`?

Reply:
Haha, yes that's a better alternative 😄

Reply:
Issue created for it - https://github.com/asyncapi/generator-model-sdk/issues/122